File: src/dto/workspace.py
from pydantic import BaseModel

from src.models.common import UserAccess


class CreateWorkspacePayload(BaseModel):
    organization_id: int
    name: str


class UpdateWorkspacePayload(BaseModel):
    workspace_id: int
    name: str


class WorkspaceSubElement(BaseModel):
    id: int
    name: str


class WorkspaceListWithAccess(BaseModel):
    access: UserAccess
    workspaces: list[WorkspaceSubElement]

File: front/src/stores/models/workspace.ts
import type { UserAccess } from "./common";

export interface WorkspaceModel {
  id: number,
  name: string,
}

export interface WorkspaceUpdateModel {
  workspace_id: number,
  name: string,
}

export interface WorkspaceForOrganizationResponse {
  access: UserAccess,
  workspaces: WorkspaceModel[],
}
